projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/components/linked-cases-table/linked-cases-to-table.component.ts

```diff
@@ -0,0 +1,164 @@
+  caseReference: string
+  caseName: string
+  caseType: string;
+  caseTypeDescription: string;
+  service: string;
+  state: string;
+  stateDescription: string;
+  reasons: []
+}
+
+@Component({
+  selector: 'ccd-linked-cases-to-table',
+  templateUrl: './linked-cases-to-table.component.html',
+  styleUrls: ['./linked-cases-to-table.component.scss']
+})
+
+export class LinkedCasesToTableComponent implements OnInit, AfterViewInit {
+
+  private static readonly CASE_CONSOLIDATED_REASON_CODE = 'CLRC015';
+  private static readonly CASE_PROGRESS_REASON_CODE = 'CLRC016';
+
+  @Input()
+  public caseField: CaseField;
+
+  @Output()
+  public notifyAPIFailure: EventEmitter<boolean> = new EventEmitter(false);
+  public caseDetails: CaseView;
+  public isLoaded: boolean;
+  public linkedCasesFromResponse: LinkedCasesResponse[] = [];
+  public caseId: string;
+  public isServerError = false;
+  public jurisdictionsResponse: Jurisdiction[] = [];
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    public readonly linkedCasesService: LinkedCasesService,
+    private  readonly casesService: CasesService) { }
+
+  public ngAfterViewInit(): void {
+    let labelField = document.getElementsByClassName('govuk-heading-l');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
+    }
+    labelField = document.getElementsByClassName('heading-h2');
+    if (labelField && labelField.length) {
+      labelField[0].replaceWith('')
+    }
+  }
+
+  public ngOnInit(): void {
+    this.caseId = this.route.snapshot && this.route.snapshot.data && this.route.snapshot.data.case.case_id;
+    this.getAllLinkedCaseInformation();
+    if (this.route.snapshot.data.case) {
+      this.linkedCasesService.caseDetails = this.route.snapshot.data.case;
+    }
+  }
+
+  public getCaseRefereneLink(caseRef) {
+    return caseRef.slice(this.caseId.length - 4);
+  }
+
+  public sortLinkedCasesByReasonCode(searchCasesResponse) {
+    const topLevelresultArray = [];
+    let secondLevelresultArray = [];
+    searchCasesResponse.forEach((item: any) => {
+      const reasons = item && item.reasons || [];
+      const consolidatedStateReason = reasons.map(x => x).find(
+        reason => reason === LinkedCasesToTableComponent.CASE_CONSOLIDATED_REASON_CODE
+      );
+      const progressedStateReason = reasons.map(x => x).find(
+        reason => reason === LinkedCasesToTableComponent.CASE_PROGRESS_REASON_CODE
+      );
+      let arrayItem;
+      if (progressedStateReason) {
+        arrayItem = { ...item };
+        topLevelresultArray.push(arrayItem);
+      } else if (consolidatedStateReason) {
+        arrayItem = { ...item };
+        secondLevelresultArray = [{ ...item }, ...secondLevelresultArray]
+      } else {
+        arrayItem = { ...item };
+        secondLevelresultArray.push({ ...item });
+      }
+    })
+    return topLevelresultArray.concat(secondLevelresultArray)
+  }
+
+  public getAllLinkedCaseInformation() {
```

Comment:
@johnbenjamin-hmcts - please specify return type if possible

Reply:
@Daniel-Lam I have updated the code now.